code/mob/living/carbon/human/procs/damage.dm

```diff
@@ -18,6 +18,22 @@
 			S.active = 0
 			S.icon_state = "shield0"
 
+	var/reflect = 0
+	var/obj/item/equipped = src.equipped()
+	if (equipped && equipped.hasProperty("reflection"))
+		reflect = 1
+	else
+		equipped = src.check_block()
+		if (equipped && equipped.hasProperty("reflection"))
+			reflect = 1
```

Comment:
Gross. Better to have reflect be an equipment property and check the associated mob property